xds/src/main/java/io/grpc/xds/FallbackManager.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static io.grpc.LoadBalancer.ATTR_LOAD_BALANCING_CONFIG;
+
+import com.google.common.collect.ImmutableList;
+import io.grpc.Attributes;
+import io.grpc.ChannelLogger.ChannelLogLevel;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.LoadBalancerRegistry;
+import io.grpc.SynchronizationContext.ScheduledHandle;
+import io.grpc.internal.ServiceConfigUtil;
+import io.grpc.xds.XdsLbState.SubchannelStore;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+final class FallbackManager {
```

Comment:
Is this going to be only used in `XdsLoadBalancer` as a final field? I don't see a benefit of making it a separate class. Usually the valid reasons for making a class is:
1. Reusability.
1. Complex implementation underneath a simple interface. A separate class would make it easy to test it separately, and its user will mock this interface in their own tests.

It's debatable whether Case 2 is the case here, but I feel it's a quite large interface (5 methods) compared to its implementation. Even if you defined an interface for it and mocked it in `XdsLoadBalancerTest`, just having the two classes unit-tested separately wouldn't provide enough confidence of them working as a whole, which calls for the need of an integration test (of the two classes). Maybe just merge it into `XdsLoadBalancer`.

Reply:
The `fallbackManager` instance in my mind will be passed to `AdsStream` response handler. Once `AdsStream` is active, it will use the `fallbackManager` to cancel fallback. And once `AdsStream` is closed and no ready subchannels, it may use the `fallbackManager` to `maybeUseFallbackPolicy()`.

Reply:
`AdsStream` only calls the 2 methods, compared to 5 methods on the interface. Alternatively, this could be done by `XdsLoadBalancer` passing a callback to `AdsStream`, which has `onWorking()` and `onClosed()`, and do the fallback handling inside `XdsLoadBalancer`.

Reply:
Done.